PatternPal/PatternPal.Core/Checks/MethodCheck.cs

```diff
@@ -1,100 +1,29 @@
 ﻿namespace PatternPal.Core.Checks;
 
 /// <summary>
-/// Checks for a method of an entity, depending on the list of <see cref="_checks"/> provided.
-/// The checks performed can be a collection of <see cref="TypeCheck"/>s, <see cref="ModifierCheck"/>s,
-/// <see cref="ParameterCheck"/>s, <see cref="RelationCheck"/>s, etc.
+/// <see cref="ICheck"/> implementation for <see cref="IMethod"/> entities.
 /// </summary>
-internal class MethodCheck : CheckBase
+internal class MethodCheck : NodeCheck< IMethod >
 {
-    //A list of checks needed to perform on the method.
-    private readonly IEnumerable<ICheck> _checks;
-
-    //A list of all found instances
-    internal List<INode> MatchedEntities { get; private set; }
-
-    /// <summary>
-    /// Initializes a new instance of the <see cref="MethodCheck"/> class.
-    /// </summary>
-    /// <param name="priority">Priority of the check.</param>
-    /// <param name="checks">A list of checks needed to perform on the method.</param>
+    /// <inheritdoc cref="NodeCheck{TNode}"/>
     internal MethodCheck(
         Priority priority,
-        IEnumerable<ICheck> checks)
-        : base(priority)
+        IEnumerable< ICheck > checks)
+        : base(
+            priority,
+            checks)
     {
-        MatchedEntities = new List<INode>();
-        _checks = checks;
     }
 
-    internal Func<List<INode>> Result => () => MatchedEntities;
-
     /// <inheritdoc />
-    public override ICheckResult Check(
+    protected override IEntity GetType4TypeCheck(
```

Comment:
Add a comment explaining that the type checked in the case of a method is the return type